contracts/contracts/Umbra.sol

```diff
@@ -1,10 +1,63 @@
 pragma solidity ^0.5.0;
 
 import "@openzeppelin/contracts-ethereum-package/contracts/ownership/Ownable.sol";
+import "@openzeppelin/contracts-ethereum-package/contracts/utils/ReentrancyGuard.sol";
+import "@openzeppelin/contracts-ethereum-package/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts-ethereum-package/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts-ethereum-package/contracts/token/ERC20/SafeERC20.sol";
 
-contract Umbra is Ownable {
+contract Umbra is Ownable, ReentrancyGuard {
+    using SafeMath for uint256;
 
-    constructor() public {
+    event EthAnnouncement(address indexed receiver, uint256 indexed amount, string note);
+    event TokenAnnouncement(address indexed receiver, uint256 indexed amount, address indexed token, string note);
```

Comment:
Should we only have one type of event announcement, that way it's a little cleaner when scanning events from the frontend? ETH Announcements can either use the zero address or `0x00eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee` which is what Kyber uses